src/tool_cb_hdr.c

```diff
@@ -103,14 +116,26 @@ size_t tool_header_cb(char *ptr, size_t size, size_t nmemb, void *userdata)
   if(per->config->etag_save_file && etag_save->stream) {
     /* match only header that start with etag (case insensitive) */
     if(curl_strnequal(str, "etag:", 5)) {
-      char *etag_start = ptr + strlen("ETag:");
+      const char *etag_start = ptr + strlen("ETag:");
+      const char *expected_etag_end, *etag_end;
+      size_t etag_length;
 
       /* pass all spaces from start */
       while(*etag_start && ISSPACE(*etag_start))
         etag_start++;
 
+      /* We expect CRLF as EOL but when called from tests it may be CR or LF */
+      expected_etag_end = end - strlen("\"\r\n");
+      if(*expected_etag_end == '"') {
+        /* step back to quote */
+        etag_end = expected_etag_end + 1;
+      }
+      else {
+        etag_end = skip_eol_at_end(end);
+      }
```

Comment:
Now this code looks really complicated and hard to follow. Isn't this just finding the last byte of the line that isn't a CR or LF?

Reply:
yes, the skip_eol_at_end have a comment. In fact it would be great to extract it to some string utilities